### PR TITLE
fix: Properly remap source_content to source_index

### LINKED ISSUE #86: `Source out of range` error if source is duplicated
This issue is mainly cause by duplicated source in a SourceMap, which I believe there is an error on the VLQMap genearted by other compiler, however the SourceMap v3 specification does not stated which as an variant.

## Reproduction
follow the step in this repository.
https://github.com/h-a-n-a/issue_1208_parcel_sourcemap_vlq

## Debug
It looks like the main issue is caused by a memoization mechanism in [this](https://github.com/parcel-bundler/source-map/blob/93c0627a91eebaad2592f7ccbc900fccd6101a4e/parcel_sourcemap/src/lib.rs#L214) function and [this](https://github.com/parcel-bundler/source-map/blob/93c0627a91eebaad2592f7ccbc900fccd6101a4e/parcel_sourcemap/src/lib.rs#L304) function throws as index is larger than the length of sources.

### PR DESCRIPTION
**Pull request by DeMoorJasper**
Apparently forgot to remap sources_content indexes to source indexes in `add_vlq_map`, surprised this didn't come up earlier.

Also found a small thing that could be improved while debugging this

Closes #86

### CODE REVIEW

File: parcel_sourcemap/src/lib.rs [hunk 1]
```diff
@@ -236,17 +236,13 @@ impl SourceMap {
         result_vec
     }
 
-    pub fn get_source_index(&self, source: &str) -> Result<Option<u32>, SourceMapError> {
+    pub fn get_source_index(&self, source: &str) -> Option<u32> {
```
Comment (devongovett) [inline parcel_sourcemap/src/lib.rs:239]: Technically a breaking API change?

**Review by devongovett — COMMENTED** [review]
Should we add a test?

File: parcel_sourcemap/src/lib.rs (hunk 1 above)
Comment (DeMoorJasper) [reply parcel_sourcemap/src/lib.rs:239]: Right... totally forgot we published this crate, doing a major version for this seems a bit much, will undo it

**Comment by DeMoorJasper** [comment]
Will add a test for it tomorrow

**Review by devongovett — APPROVED** [review]

**Comment by h-a-n-a** [comment]
@DeMoorJasper I currently have a project rely on this. Would you mind making a release? :)

**Pull request merged**